Client/src/Pages/Monitors/Details/Charts/index.jsx

```diff
@@ -0,0 +1,280 @@
+import { useTheme } from "@emotion/react";
+import {
+  BarChart,
+  Bar,
+  XAxis,
+  CartesianGrid,
+  ResponsiveContainer,
+  Cell,
+  RadialBarChart,
+  RadialBar,
+} from "recharts";
+import { formatDate } from "../../../../Utils/timeUtils";
+import { useState } from "react";
+
+const CustomLabels = ({
+  x,
+  width,
+  height,
+  firstDataPoint,
+  lastDataPoint,
+  type,
+}) => {
+  let options = {
+    month: "short",
+    year: undefined,
+    hour: undefined,
+    minute: undefined,
+  };
+  if (type === "day") delete options.hour;
+
+  return (
+    <>
+      <text x={x} y={height} dy={-3} textAnchor="start" fontSize={11}>
+        {formatDate(new Date(firstDataPoint.time), options)}
+      </text>
+      <text x={width} y={height} dy={-3} textAnchor="end" fontSize={11}>
+        {formatDate(new Date(lastDataPoint.time), options)}
+      </text>
+    </>
+  );
+};
+
+export const UpBarChart = ({ data, type, onBarHover }) => {
+  const theme = useTheme();
+
+  const [chartHovered, setChartHovered] = useState(false);
+  const [hoveredBarIndex, setHoveredBarIndex] = useState(null);
+
+  const getColorRange = (uptime) => {
+    return uptime > 80
+      ? { main: theme.palette.success.main, light: theme.palette.success.light }
+      : uptime > 50
+      ? { main: theme.palette.warning.main, light: theme.palette.warning.light }
+      : { main: theme.palette.error.text, light: theme.palette.error.light };
+  };
+
+  // TODO - REMOVE THIS LATER
+  let reversedData = [...data].reverse();
+
+  return (
+    <ResponsiveContainer width="100%" height={155}>
+      <BarChart
+        width="100%"
+        height="100%"
+        data={reversedData}
+        onMouseEnter={() => {
+          setChartHovered(true);
+          onBarHover({ time: null, totalChecks: 0, uptimePercentage: 0 });
+        }}
+        onMouseLeave={() => {
+          setChartHovered(false);
+          setHoveredBarIndex(null);
+          onBarHover(null);
+        }}
+      >
+        <XAxis
+          stroke={theme.palette.border.dark}
+          height={15}
+          tick={false}
+          label={
+            <CustomLabels
+              x={0}
+              y={0}
+              width="100%"
+              height="100%"
+              firstDataPoint={reversedData[0]}
+              lastDataPoint={reversedData[reversedData.length - 1]}
+              type={type}
+            />
+          }
+        />
+        <Bar
+          dataKey="totalChecks"
+          maxBarSize={7}
+          background={{ fill: "transparent" }}
+        >
+          {reversedData.map((entry, index) => {
+            let { main, light } = getColorRange(entry.uptimePercentage);
+            return (
+              <Cell
+                key={`cell-${index}`}
```

Comment:
We should use a unique identifier here, adding the `cell` prefix doesnt bamboozle react :joy: Is there any unique data here?  I can't remember :joy: